### PR TITLE
Add a way to store additional data in GLTFState and GLTFNode

### PR DESCRIPTION
**Pull request by aaronfranke**
This PR is a subset of #66026 for easier reviewing.

This PR allows GLTFDocumentExtension classes to be stateless by adding a way to store additional data in GLTFState and GLTFNode. This uses a Dictionary internally, with each GLTFDocumentExtension class storing arbitrary data with their name as the key. Therefore, it's only exposed via wrapper methods to ensure the API is used correctly.

This work was sponsored by The Mirror.

### CODE REVIEW

File: modules/gltf/structures/gltf_node.h [hunk 1]
```diff
@@ -96,6 +97,10 @@ class GLTFNode : public Resource {
 
 	GLTFLightIndex get_light();
 	void set_light(GLTFLightIndex p_light);
+
+	Variant get_additional_data(const String &p_extension_name);
```
Comment (fire) [inline modules/gltf/structures/gltf_node.h:101]: Does it matter if it's string or stringname?

File: modules/gltf/structures/gltf_node.cpp [hunk 2]
```diff
@@ -176,3 +179,15 @@ GLTFLightIndex GLTFNode::get_light() {
 void GLTFNode::set_light(GLTFLightIndex p_light) {
 	light = p_light;
 }
+
+Variant GLTFNode::get_additional_data(const String &p_extension_name) {
+	return additional_data[p_extension_name];
```
Comment (fire) [inline modules/gltf/structures/gltf_node.cpp:184]: What happens if additional_data does not container extension name?

File: modules/gltf/gltf_state.cpp [hunk 3]
```diff
@@ -358,3 +361,15 @@ String GLTFState::get_base_path() {
 void GLTFState::set_base_path(String p_base_path) {
 	base_path = p_base_path;
 }
+
+Variant GLTFState::get_additional_data(const String &p_extension_name) {
+	return additional_data[p_extension_name];
```
Comment (fire) [inline modules/gltf/gltf_state.cpp:366]: Same here too.

File: modules/gltf/structures/gltf_node.h (hunk 1 above)
Comment (aaronfranke) [reply modules/gltf/structures/gltf_node.h]: Either works as long as it's consistent. I think that StringName is better. Changed.

File: modules/gltf/structures/gltf_node.cpp (hunk 2 above)
Comment (aaronfranke) [reply modules/gltf/structures/gltf_node.cpp:183]: It returns null, which actually makes me realize, we don't need `has_additional_data`. I've documented this behavior and removed `has_additional_data`.

File: modules/gltf/gltf_state.cpp (hunk 3 above)
Comment (aaronfranke) [reply modules/gltf/gltf_state.cpp:365]: It returns null, which actually makes me realize, we don't need `has_additional_data`. I've documented this behavior and removed `has_additional_data`.

**Review by fire — APPROVED** [review]

**Pull request merged**